File: arr.go
package main

import "fmt"

func checkSum(arr []int, x int) bool {
    seen := make(map[int]bool)

    for _, num := range arr {
        complement := x - num
        if seen[complement] {
            return true
        }
        seen[num] = true
    }

    return false
}

func main() {
    arr := []int{0, -1, 42, -3, 1}
    x := -2

    if checkSum(arr, x) {
        fmt.Println("Two elements exist in the array whose sum is exactly", x)
    } else {
        fmt.Println("No two elements exist in the array whose sum is exactly", x)
    }
}

File: arr.js
function checkSum(arr, x) {
    let map = {};
    for (let i = 0; i < arr.length; i++) {
        if (map[arr[i]]) {
            return true;
        }
        map[x - arr[i]] = true;
    }
    return false;
}

let arr = [0, -1, 42, -3, 1];
let x = -2;
console.log(checkSum(arr, x));
